### PR TITLE
Math exceptions

### PR DESCRIPTION
**Pull request by baykaner**

### CODE REVIEW

File: libs/math/include/math/fundamental_operators.hpp [hunk 1]
```diff
@@ -21,6 +21,7 @@
 #include "math/tensor_broadcast.hpp"
 
 #include <cassert>
+#include <math/exceptions/exceptions.hpp>
```
Comment (evsmithx) [inline libs/math/include/math/fundamental_operators.hpp]: <> -> ""

File: libs/math/include/math/matrix_operations.hpp [hunk 2]
```diff
@@ -1063,7 +1064,7 @@ fetch::math::meta::IfIsMathArray<ArrayType, void> Dot(ArrayType const &A, ArrayT
 
   if (aview.width() != bview.height())
   {
-    throw std::runtime_error("expected A width to equal and B height.");
+    throw fetch::math::exceptions::WrongShape("expected A width to equal and B height.");
```
Comment (evsmithx) [inline libs/math/include/math/matrix_operations.hpp]: "expected A width to equal B height" maybe?

**Review by evsmithx — CHANGES_REQUESTED** [review]
Just wondering - where and how do these exceptions get handled by Etch?

**Review by evsmithx — DISMISSED** [review]
LGTM

**Review by MissingNO57 — APPROVED** [review]
LGTM

**Pull request merged**